metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -228,20 +213,14 @@ class ScalametaLanguageServer(
         ()
     }
 
-  override def onChangeConfiguration(settings: JsValue): Unit = {
-    (settings \ "scalameta").validate[Configuration] match {
-      case JsSuccess(value, _) =>
-        logger.info(s"Configuration changed $value")
-      case JsError(error) =>
-        logger.error(s"Can't decode configuration: $error")
-    }
-  }
+  override def onChangeConfiguration(settings: JsValue): Unit =
+    configurationSubscriber.onNext(settings)
```

Comment:
What does this actually do?  I poked around the code and couldn’t figure it out.

Reply:
`onChangeConfiguration` is invoked whenever the client updates the settings. This line publishes the new value onto the Observable.

Reply:
Just tracked that down.  Thanks for the confirmation.

Reply:
Ideally we will change the implementation to directly use Observables for notifications.

I.e. instead of having callbacks, we would just subscribe to the appropriate observables.

Reply:
👍